packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogSdkPlugin.kt

```diff
@@ -200,6 +215,84 @@ class DatadogSdkPlugin : FlutterPlugin, MethodCallHandler {
         isRegistered.set(false)
     }
 
+    @Suppress("TooGenericExceptionCaught")
+    private fun mapLogEvent(event: LogEvent): LogEvent? {
+        val jsonEvent = event.toJson().asMap()
+
+        var modifiedJson: Map<String, Any?>? = null
+
+        val semaphore = Semaphore(1)
+        semaphore.acquire()
+
+        val handler = Handler(Looper.getMainLooper())
+        handler.post {
+            try {
+                channel.invokeMethod(
+                    "mapLogEvent",
+                    mapOf(
+                        "event" to jsonEvent
+                    ),
+                    object : MethodChannel.Result {
+                        override fun success(result: Any?) {
+                            modifiedJson = result as? Map<String, Any?>
+                            semaphore.release()
+                        }
+
+                        override fun error(
+                            errorCode: String,
+                            errorMessage: String?,
+                            errorDetails: Any?
+                        ) {
+                            // No telemetry needed, this is likely an issue in user code
+                            semaphore.release()
+                        }
+
+                        override fun notImplemented() {
+                            Datadog._internal._telemetry.error(
+                                "mapLogEvent returned notImplemented."
+                            )
+                            semaphore.release()
+                        }
+                    }
+                )
+            } catch (t: Throwable) {
```

Comment:
should we narrow down it to `Exception`? Because if it is `Error`, there is something wrong with the runtime itself and we shouldn't prevent it from crashing (`Error` mostly likely will crash everything, unlikely that the code up in the stack will catch it).

But anyway, I'm ok with `Throwable` as well (as I understand it is so wide to release semaphore in any case).

Reply:
🤔 You may be right. I'm okay switching to Exception as that should probably catch anything that's recoverable anyway.